File: pset6/sentimental-mario-more/mario.js
const readline = require('readline');


// Set height bounds
const MAX_HEIGHT = 8;
const MIN_HEIGHT = 1;


// Create readline interface
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});


// Prompt for height until valid input (Between min and max)
function promptHeight() {
    rl.question('Who tall boss: ', answer => {
        const height = parseInt(answer, 10);
        if (isNaN(height) || height < MIN_HEIGHT || height > MAX_HEIGHT) {
            // invalid input -> ask again
            promptHeight();
        } else {
            buildPyramid(height);
            rl.close();
        }
    });
}


// Build and print the pyramid
function buildPyramid(height){
    for (let row = 1; row <= height; row++) {
        const spaces = ' '.repeat(height - row); // Spaces
        const blocks = '#'.repeat(row); //Hashes for row
        console.log(`${spaces}${blocks}  ${blocks}`) //Building with two spaces
    }
}


// Start
promptHeight();

File: pset6/sentimental-mario-more/mario.py
def main():
    # set height bounds
    MAX_HEIGHT = 8
    MIN_HEIGHT = 1

    # Prompt for height until valid input
    height = 0
    while height < MIN_HEIGHT or height > MAX_HEIGHT:
        try:
            height = int(input("Who tall boss: "))
        except ValueError:
            height = 0

    # Build and print pyramid
    for row in range(1, height + 1):
        spaces = height - row
        blocks = "#" * row
        print(" " * spaces + blocks + "  " + blocks)


if __name__ == "__main__":
    main()
